src/sensor_icm20948.c

```diff
@@ -23,8 +23,8 @@
 
 #define FIFO_OVERFLOW_INT 0x0F // from datasheet
 
-#define BYTES_PER_FIFO_ENTRY 6
-#define BYTES_PER_BLOCK 48
+#define BYTES_PER_SAMPLE 6
+#define BYTES_PER_BLOCK (BYTES_PER_SAMPLE * 2)
```

Comment:
I'm not an expert on the icm20948, but this looks odd.  What is the reason for decreasing the data read size?

Reply:
This is my attempt to decrease the probability of a partial read, fewer bytes per attempt -> less probability, I assume.
I'm concerned that if we do not fail the system, then it is possible to read less than 6 bytes of data.
So, all the next bytes will probably be shifted by that amount, so all data readed after that will be borked.
Miss 1 byte - borked.
Miss 2 bytes - axis swapped.

Reply:
FWIW, if there are i2c errors, the data is borked - nothing we can do about that other than tell users they need to have clean i2c lines to meaningfully use the accelerometer.  So, reporting warnings/errors/overflows is better than a shutdown, but I don't think we can reasonably get "good" data from the accelerometer if there i2c errors.

I don't think smaller transfers reduces the rate of partial transfers as there would be many more i2c transfers that could fail.  (Unless the icm20948 does something weird that I'm not aware of.)

Cheers,
-Kevin